sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ClientRetryPolicy.java

```diff
@@ -177,7 +177,7 @@ private ShouldRetryResult shouldRetryOnSessionNotAvailable() {
                     // we have already tried this request on the write location
                     return ShouldRetryResult.noRetry();
                 } else {
-                    this.retryContext = new RetryContext(this.sessionTokenRetryCount - 1, false);
```

Comment:
This is changing the behavior on Single master, right? It should always go to region 0 (main region). 

`this.sessionTokenRetryCount` equals 1, so 1 - 1 = 0. If we remove the -1, then won't that try to go to region 1 (if there is any)?

Reply:
We have second parameter as false, so we will go always to master region , I changed sessionTokenRetryCount-1 to this.sessionTokenRetryCount , just to be consistent with other cases, no effect of this. Test is covering single master

Reply:
As per the offline conversation sessionTokenRetryCount or sessionTokenRetryCount -1 will have same result as retryRequestOnPreferredLocations is false making it to go to master region anyway. But still we agreed to make is 0 , it will helpful in code readability that in single master we are retrying explicitly going to master.